.github/workflows/generate-changelog.yml

```diff
@@ -0,0 +1,30 @@
+name: Generate CHANGELOG
+on:
+  pull_request:
+    types: [closed]
+  workflow_dispatch:
+jobs:
+  GenerateChangelog:
+    if: github.event.pull_request.merged || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+        with:
+          go-version-file: 'go.mod'
+      - run: make tools update-changelog-unreleased-section
+      - run: |
+          if [[ $(git status --porcelain) ]]; then
+            if ${{github.event_name == 'workflow_dispatch'}}; then
+              MSG="Update CHANGELOG.md (Manual Trigger)"
+            else
+              MSG="Update CHANGELOG.md for #${{ github.event.pull_request.number }}"
+            fi
+            git config --local user.email changelogbot@hashicorp.com
```

Comment:
we might want to use a MongoDB address

Reply:
I used the [same](https://github.com/hashicorp/terraform-provider-aws/blob/60cf7fd65c391de1e8fc200207131eb0f35fe028/.github/workflows/generate_changelog.yml#L26) as AWS uses in their provider. Happy to change it. 
Would `changelogbot@mongodb.com` be better?

Reply:
not sure what email it's best to use, maybe @gssbzn can help.
as an example I found in CLI:
maintainer: MongoDB Inc. <support@mongodb.com>

Reply:
We have cloud bot (https://github.com/mms-build-account) but you have to consider in the near feature all shipped product needs it's commit signed according to SSDLC cc @marcosuma

Reply:
We can make use of ghaction-import-gpg action to sign commits https://github.com/actions/runner/issues/667#issuecomment-775277257. We currently use this action for signing our release. @oarbusi We can include in a followup PR but seems reasonable.

Regarding
> Would changelogbot@mongodb.com be better?

I agree.